spotlight-client/src/metricsApi/fetchMetrics.ts

```diff
@@ -30,6 +30,7 @@ type ErrorAPIResponse = { error: string };
 type FetchMetricOptions = {
   metricNames: string[];
   tenantId: TenantId;
+  getToken: () => Promise<string>;
```

Comment:
what was the issue here, importing the datastore into this module was causing some kind of dependency resolution problem? I wonder if it might be better solved by moving the API methods into a datastore class rather than drilling this function through all of our content models? (I don't know off the top of my head if that would cause different problems but it seems like it might be cleaner, if it works)

Reply:
Yeah I was going to ask here, if it would be easy to move the API methods into one of our stores and call it without having to pass it down all these content models?

Reply:
Because I was getting nasty dependency cycles with trying to import the Datastore in fetchmetrics

Reply:
I think you should give it a try! IIRC I kept them separate because at the time they didn't really have any dependencies, but there isn't any structural reason why they _have to_ be set up that way